src/sensor.cpp

```diff
@@ -347,14 +347,28 @@ namespace librealsense
                     int width = vsp ? vsp->get_width() : 0;
                     int height = vsp ? vsp->get_height() : 0;
 
-                    frame_holder fh = _source.alloc_frame(stream_to_frame_types(req_profile_base->get_stream_type()), width * height * bpp / 8, fr->additional_data, requires_processing);
+                    assert((width * height) % 8 == 0);
+                    frame_holder fh = _source.alloc_frame(stream_to_frame_types(req_profile_base->get_stream_type()), (width * height * bpp) >> 3, fr->additional_data, requires_processing);
                     auto diff = environment::get_instance().get_time_service()->get_time() - system_time;
                     if (diff >10 )
                         LOG_DEBUG("!! Frame allocation took " << diff << " msec");
 
                     if (fh.frame)
                     {
-                        memcpy((void*)fh->get_frame_data(), fr->data.data(), sizeof(byte)*fr->data.size());
+                        auto expected_size = width * height * (float)bpp / 8.f;
+
+                        if (expected_size != sizeof(byte)*fr->data.size())
```

Comment:
As discussed -- this needs changing

Reply:
changed

Reply:
Change to assert now that the 5M frame size is as expected

Reply:
done